kafka-impl/src/main/java/io/streamnative/pulsar/handlers/kop/KopEventManager.java

```diff
@@ -288,54 +289,74 @@ void process(BiConsumer<String, Long> registerEventLatency,
-                    kopTopicsSet.add(kopTopic);
-                    topicsFullNameDeletionsSets.add(kopTopic.getFullName());
-                });
-
                 if (log.isDebugEnabled()) {
                     log.debug("Delete topics listener fired for topics {} to be deleted", topicsDeletions);
                 }
 
-                Iterable<GroupMetadata> groupMetadataIterable = coordinator.getGroupManager().currentGroups();
-                HashSet<TopicPartition> topicPartitionsToBeDeletions = Sets.newHashSet();
-
-                groupMetadataIterable.forEach(groupMetadata -> {
-                    topicPartitionsToBeDeletions.addAll(
-                            groupMetadata.collectPartitionsWithTopics(topicsFullNameDeletionsSets));
-                });
-
-                Set<String> deletedTopics = Sets.newHashSet();
-                if (!topicPartitionsToBeDeletions.isEmpty()) {
-                    coordinator.handleDeletedPartitions(topicPartitionsToBeDeletions);
-                    Set<String> collectDeleteTopics = topicPartitionsToBeDeletions
-                            .stream()
-                            .map(TopicPartition::topic)
-                            .collect(Collectors.toSet());
-
-                    deletedTopics = kopTopicsSet.stream().filter(
-                            kopTopic -> collectDeleteTopics.contains(kopTopic.getFullName())
-                    ).map(KopTopic::getOriginalName).collect(Collectors.toSet());
-
-                    deletedTopics.forEach(deletedTopic -> {
-                        metadataStore.delete(
-                                getDeleteTopicsPath() + "/" + deletedTopic, Optional.of((long) -1));
-                    });
-                }
+                // Localize groupCoordinatorsByTenant to avoid multi-thread conflicts
+                final Map<String, GroupCoordinator> currentCoordinators = groupCoordinatorsByTenant;
+                final Set<String> deletedTopics = Sets.newConcurrentHashSet();
+                final AtomicInteger pendingCoordinators = new AtomicInteger(currentCoordinators.size());
+
+                currentCoordinators.forEach((tenant, groupCoordinator) -> {
+
+                    // TODO
+                    // now we have tenant/coordinator/topicsDeletions
+                    // It is possible that different coordinators have the same Kafka topics?
+                    // Because there is no multi-tenancy on the Kafka side
+                    // The question here is, which topic does the kafka admin client delete?
+                    // Is it related to tenants or delete all kafka topics with the same name under all tenants?
+                    // The implementation here only follows the concept that there is no multi-tenancy in Kafka,
+                    // that mean deleted the Kafka topics with the same name under all tenants,
+                    // and leave the rest for everyone to discuss
```

Comment:
leave a discuss

Reply:
> which topic does the kafka admin client delete?

There're two configs `kafkaTenant` and `kafkaNamespace` that represent the default namespace in `KopTopic`.

Does it answer your questions? If yes, I think these comments can be removed.

Reply:
Yes. Let us remove these comments. :)